File: examples/nim-api/math.nim
import std/strutils

proc double(x: int): int =
  x * 2

proc add(a, b: int): int =
  a + b

type
  Point = object
    x, y: int

func distance(a, b: Point): float =
  sqrt(float((a.x - b.x)^2 + (a.y - b.y)^2))

File: go/internal/index/testdata/langexp2/fsharp/Math.fs
module Math

open System

let double x = x * 2

let add a b = a + b

type Point = { x: int; y: int }

let origin = { x = 0; y = 0 }

let distance (a: Point) (b: Point) =
    let dx = a.x - b.x
    sqrt (float (dx * dx))
